neurogym/envs/contextdecisionmaking.py

```diff
@@ -168,55 +149,40 @@ def _new_trial(self, **kwargs: Any) -> dict[str, Any]:  # type: ignore[override]
         # Set observations based on context type
         self.add_ob(1, where="fixation")
 
+        self._set_observations(choice_1, choice_2, coh_1, coh_2)
+        # Add context signals for explicit context
         if self.use_expl_context:
-            self._set_expl_observations(choice_1, choice_2, coh_1, coh_2)
-            # Add context signals
             if trial["context"] == 0:
                 self.add_ob(1, where="context1")
             else:
                 self.add_ob(1, where="context2")
-        else:
-            self._set_impl_observations(choice_1, choice_2, coh_1, coh_2)
         self.set_groundtruth(trial["ground_truth"], "decision")
 
         return trial
 
-    def _set_expl_observations(self, choice_1, choice_2, coh_1, coh_2):
-        """Set observations for direct value representation (explicit context)."""
-        signed_coh_1 = coh_1 if choice_1 == 1 else -coh_1
-        signed_coh_2 = coh_2 if choice_2 == 1 else -coh_2
-
-        self.add_ob((1 + signed_coh_1 / 100) / 2, period="stimulus", where="stim1_mod1")
-        self.add_ob((1 - signed_coh_1 / 100) / 2, period="stimulus", where="stim2_mod1")
-        self.add_ob((1 + signed_coh_2 / 100) / 2, period="stimulus", where="stim1_mod2")
-        self.add_ob((1 - signed_coh_2 / 100) / 2, period="stimulus", where="stim2_mod2")
-
-        self.add_randn(0, self.sigma, "stimulus")
-        self.set_ob(0, "decision")
-
-    def _set_impl_observations(self, choice_1, choice_2, coh_1, coh_2):
-        """Set observations for ring representation (implicit context)."""
+    def _set_observations(self, choice_1, choice_2, coh_1, coh_2):
+        """Set observations for ring representation."""
         stim_theta_1 = self.theta[choice_1 - 1]
         stim_theta_2 = self.theta[choice_2 - 1]
 
         stim_mod1 = np.cos(self.theta - stim_theta_1) * (coh_1 / 200) + 0.5
         stim_mod2 = np.cos(self.theta - stim_theta_2) * (coh_2 / 200) + 0.5
 
         # Add observations for each index
-        for i in range(self.impl_dim_ring):
-            self.add_ob(stim_mod1[i], "stimulus", where=f"stimulus{i + 1}_mod1")
-            self.add_ob(stim_mod2[i], "stimulus", where=f"stimulus{i + 1}_mod2")
+        for i in range(self.dim_ring):
+            self.add_ob(stim_mod1[i], "stimulus", where=f"stim{i + 1}_mod1")
+            self.add_ob(stim_mod2[i], "stimulus", where=f"stim{i + 1}_mod2")
```

Comment:
why was this abbreviated?

Reply:
Because of how it's defined in `obs_space_name`:

```python
        obs_space_name = {
            "fixation": 0,
            **{f"stim{i + 1}_mod1": i + 1 for i in range(self.dim_ring)},
            **{f"stim{i + 1}_mod2": i + 1 + self.dim_ring for i in range(self.dim_ring)},
        }
```